src/helpers/availabilitiesList.ts

```diff
@@ -0,0 +1,16 @@
+import {Availability} from "../models/doctor"
+import apiJoovence, {availabilitiesMap} from "../conf/api.joovence"
+
+export const availabilitiesList = (id:string, setAvailabilities:(value:Availability[])=> void) => {
+    // Fetch availabilities list
+    const getAvailabilities = async () => {
+        try{
+            const response = await apiJoovence.get(`availabilities?doctorId=${id}`);
+            const availabilitiesList = await response.data.map(availabilitiesMap);
+            setAvailabilities(availabilitiesList);
+        }catch(error){
+            console.log(error);
+        }
+    };
+    getAvailabilities();
+};
```

Comment:
Again, sending the `setAvailabilities` involves creating too much dependencies and makes it difficult to reuse. Instead you should return the results and handle in the component.
```suggestion
export const availabilitiesList = async (id:string): Promise<Availability[]> => {
    // Fetch availabilities list
        try {
            const response = await apiJoovence.get(`availabilities?doctorId=${id}`);
            const availabilities = await response.data.map(availabilitiesMap);
            return availabilities;
        } catch(error) {
            console.log(error);
            return []
        }
    };
};
```